src/payment/strategies/digitalriver/digitalriver-payment-strategy.ts

```diff
@@ -72,35 +74,81 @@ export default class DigitalRiverPaymentStrategy implements PaymentStrategy {
             this._unsubscribe();
         }
 
+        const { containerId } = this._getDigitalRiverInitializeOptions();
+        const container = document.getElementById(containerId);
+
+        if (container) {
+            container.innerHTML = '';
+        }
+
         return Promise.resolve(this._store.getState());
     }
 
     async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        const { payment, ...order } = orderRequest;
-        const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
+        const {payment, ...order} = orderRequest;
+
+        if (!payment || !payment.paymentData) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        const { paymentData, methodId } = payment;
+        const { shouldSetAsDefaultInstrument = false } = paymentData as HostedInstrument;
+        const {isStoreCreditApplied: useStoreCredit} = this._store.getState().checkout.getCheckoutOrThrow();
 
         if (useStoreCredit !== undefined) {
             await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
         }
 
         await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
 
-        if (!payment || !this._loadSuccessResponse || !this._digitalRiverCheckoutData) {
+        if (!this._digitalRiverCheckoutData) {
             throw new InvalidArgumentError('Unable to proceed because payload payment argument is not provided.');
```

Comment:
Keep consistency with line 230
```suggestion
            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
```